homeassistant/components/telegram_bot/__init__.py

```diff
@@ -51,44 +198,296 @@ def async_setup_platform(p_type, p_config=None, discovery_info=None):
             return
 
         _LOGGER.info("Setting up %s.%s", DOMAIN, p_type)
+        try:
+            receiver_service = yield from \
+                platform.async_setup_platform(hass, p_config, discovery_info)
+            if receiver_service is None:
+                _LOGGER.error(
+                    "Failed to initialize telegram bot %s", p_type)
+                return
+
+        except Exception:  # pylint: disable=broad-except
+            _LOGGER.exception('Error setting up platform %s', p_type)
+            return
 
+        notify_service = None
         try:
-            if hasattr(platform, 'async_setup_platform'):
-                notify_service = yield from \
-                    platform.async_setup_platform(hass, p_config,
-                                                  discovery_info)
-            elif hasattr(platform, 'setup_platform'):
-                notify_service = yield from hass.loop.run_in_executor(
-                    None, platform.setup_platform, hass, p_config,
-                    discovery_info)
-            else:
-                raise HomeAssistantError("Invalid telegram bot platform.")
+            notify_service = yield from \
+                async_get_service(hass, p_config, discovery_info)
 
             if notify_service is None:
                 _LOGGER.error(
-                    "Failed to initialize telegram bot %s", p_type)
+                    "Failed to initialize notification services")
                 return
 
         except Exception:  # pylint: disable=broad-except
-            _LOGGER.exception('Error setting up platform %s', p_type)
+            _LOGGER.exception('Error setting up notification services')
             return
 
-    setup_tasks = [async_setup_platform(p_type, p_config) for p_type, p_config
-                   in config_per_platform(config, DOMAIN)]
+        notify_service.hass = hass
```

Comment:
This is a leftover from the notify service, you can now pass `hass` in if you need it.